### ISSUE TITLE
False-Negative - unexpected-keyword-arg - using decorator with keyword arguments

### ISSUE THREAD

**Issue by pcharmoille** [PyCQA/pylint#7946]
### Bug description

```a.py
def my_decorator(function):
    from functools import wraps

    @wraps(function)
    def wrapper(*args, **kwargs):
        return function(*args, **kwargs)

    return wrapper


@my_decorator
def a(c=0):
    pass


a(d=2)
```

### Command used

```shell
pylint -E a.py
```

### Pylint output

```shell
None
```

### Expected behavior

> ************* Module test_pylint
> E: 16, 0: Unexpected keyword argument 'd' in function call (unexpected-keyword-arg)

well reported on : 
```
No config file found, using default configuration
pylint 1.9.5,
astroid 1.6.6
Python 2.7.18 (v2.7.18:8d21aa21f2, Apr 20 2020, 13:25:05) [MSC v.1500 64 bit (AMD64)]
```

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

windows amd 64b

**Comment by clavedeluna** [comment]
I believe this is the same issue as https://github.com/PyCQA/pylint/issues/5784 but I'll let a maintainer close as a clone just to get another pair of eyes on it.

**Comment by Pierre-Sassoulas** [comment]
Decorator can modify function signature in really complex way that pylint won't understand right now. We're favoring false negatives over false positives so it makes sense to not report in this case in the current state of what astroid can understand of decorators. I'm on mobile so I don't know what the issue we should close in favor of is.
